Add Tuya motion sensor `_TZE200_agumlajc` quirk

ausfas opened the pull request:
## Proposed change
This PR adds a custom quirk for the Mercator Ikuü TS0601 Motion Sensor (_TZE200_agumlajc), which is a Tuya-based combination sensor.

The quirk enables full functionality for:
- Occupancy (motion) detection using DP104
- Temperature (DP1)
- Humidity (DP2)
- Illuminance (DP101)
- Motion timeout / fade time setting
- Sensitivity, distance, and other optional datapoints

The quirk was developed using the TuyaQuirkBuilder framework, and tested successfully with ZHA in Home Assistant on firmware `0x00000042`.

## Additional information
This device previously paired but was non-functional under ZHA without a quirk. The implementation uses datapoint references confirmed via Zigbee2MQTT and verified manually through ZHA logs and testing.

This is a non-breaking addition and does not interfere with other Tuya TS0601-based devices.

Let me know if naming or structure adjustments are needed to align with ZHA device handler conventions.

## Checklist
- [x] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works



Related: #2993

ausfas committed:
Add quirk for TS0601 Motion Sensor (_TZE200_agumlajc)

ausfas committed:
Format ts0601_motion.py using black

pre-commit-ci[bot] committed:
Apply pre-commit auto fixes

ausfas committed:
Add missing docstrings to pass Ruff checks

pre-commit-ci[bot] committed:
Apply pre-commit auto fixes

ausfas committed:
Add module-level docstring to satisfy Ruff

ausfas committed:
Fix fallback name casing for motion_sensitivity translation key

TheJulianJES commented on zhaquirks/tuya/ts0601_motion.py:
```diff
@@ -0,0 +1,77 @@
+        max_value=1000,
+        step=1,
+        translation_key="brightness_threshold",
+        fallback_name="Brightness Threshold",
```
Due to HA naming conventions, please only capitalize the first word, i.e. this should be "Brightness threshold" for the `fallback_name` (which will be the basis for US English translations when it's merged into HA).
Please also change the other names.

TheJulianJES commented on zhaquirks/tuya/ts0601_motion.py:
```diff
@@ -0,0 +1,77 @@
+        translation_key="motion_hold_time",
+        fallback_name="Motion Hold Time",
+    )
+    .tuya_binary_sensor(
+        dp_id=104,
+        attribute_name="motion",
+        device_class=BinarySensorDeviceClass.MOTION,
+        entity_type=EntityType.STANDARD,
+        translation_key="motion",
+        fallback_name="Motion",
+    )
```
This is not wrong, but instead of directly mapping a Tuya dp to a v2 quirk entity, we've generally mapped the dp to a (fake) ZCL attribute on the `IasZone` cluster or the `OccupancySensing` cluster using  `TuyaOccupancySensing` or `TuyaMotionWithReset` in `tuya_motion.py`, then mapping the dp with `tuya_dp`. See the examples in that file.
We may want to follow that here as well (and/or even introduce a `tuya_motion` method at some point instead).

TheJulianJES reviewed (commented):
I think we should put this in the `tuya_motion.py` file instead.
We've generally started using the `tuya_` prefix for files containing Tuya v2 quirks.
Other similar motion quirks are already in that file as well.

And `TS0601` is used by basically all Tuya devices unfortunately (from dimmers to thermostats, ...).

pre-commit-ci[bot] committed:
Apply pre-commit auto fixes

ausfas committed:
Fix incorrect import of 'tuya' from builder

ausfas committed:
Refactor TS0601 motion quirk to use high-level tuya_binary_sensor, remove unused value_converter

pre-commit-ci[bot] committed:
Apply pre-commit auto fixes

ausfas committed:
Fix translation key and quirk registration conflicts for TS0601 motion sensor

ausfas commented:
Thanks @TheJulianJES! I’ve:

Moved the quirk into tuya_motion.py alongside similar Tuya motion devices

Replaced low-level .tuya_dp() with .tuya_binary_sensor() for motion DP (104) to align with other Tuya v2 quirks

Standardized fallback names to use sentence-style capitalization

Let me know if anything further is needed.
![image](https://github.com/user-attachments/assets/e7366d9c-e0c7-4b43-bd68-dbec8457f711)

ausfas commented:
Hi @TheJulianJES, do I need to do anything at my end or will you merge this into main branch later ?

ausfas commented:
Hi @TheJulianJES , when will you add that to the main branch ?

ausfas commented:
Hi @TheJulianJES, its been been in this state for over 3 months, could you please merge in the main branch ?

TheJulianJES commented on zhaquirks/tuya/tuya_motion.py:
```diff
@@ -1610,3 +1610,71 @@ class TuyaMotionDetectionMode(t.enum8):
+        min_value=1,
+        max_value=3600,
+        step=1,
+        translation_key="motion_hold_time",
+        fallback_name="Motion hold time",
```
We generally use `fading_time` for this apparently, not `motion_hold_time`.
Just try to match other Tuya motion quirks in terms of names, so we don't need to add even more strings/translations.

TheJulianJES commented on zhaquirks/tuya/tuya_motion.py:
```diff
@@ -1610,3 +1610,71 @@ class TuyaMotionDetectionMode(t.enum8):
+        attribute_name="motion",
+        device_class=BinarySensorDeviceClass.MOTION,
+        entity_type=EntityType.STANDARD,
+        translation_key="motion",
```
The translation key can be removed here, as a device class with "Motion" is assigned.
That will also use "Motion" as the name.

TheJulianJES commented on zhaquirks/tuya/tuya_motion.py:
```diff
@@ -1610,3 +1610,71 @@ class TuyaMotionDetectionMode(t.enum8):
+        dp_id=102,
+        attribute_name="brightness_threshold",
+        type=t.uint16_t,
+        device_class=SensorDeviceClass.ILLUMINANCE,
```
`SensorDeviceClass.ILLUMINANCE` requires `lx` (LUX) as the `unit`. This is not set here and will cause warnings in HA logs otherwise.
Reference: https://developers.home-assistant.io/docs/core/entity/sensor/

TheJulianJES commented on zhaquirks/tuya/tuya_motion.py:
```diff
@@ -1610,3 +1610,71 @@ class TuyaMotionDetectionMode(t.enum8):
+    .tuya_temperature(dp_id=1, scale=10)
+    .tuya_humidity(dp_id=2)
+    .tuya_illuminance(dp_id=101)
+    .tuya_number(
+        dp_id=102,
+        attribute_name="brightness_threshold",
+        type=t.uint16_t,
+        device_class=SensorDeviceClass.ILLUMINANCE,
+        min_value=0,
+        max_value=1000,
+        step=1,
+        translation_key="brightness_threshold",
+        fallback_name="Brightness threshold",
+    )
```
Can you also quickly explain what this entity does? Is motion not detected if it's too dark or bright?

TheJulianJES reviewed (commented):
Also kind of see the feedback I gave in this PR: https://github.com/zigpy/zha-device-handlers/pull/4044#pullrequestreview-3306640316
Related in terms of naming.

ausfas committed:
Align Mercator TS0601 quirk with reviewer feedback and move to tuya_motion.py

ausfas commented:
I’ve updated the quirk in `zhaquirks/tuya/tuya_motion.py` to align with the Tuya v2 conventions and the existing Tuya motion patterns.

Changes included:

* Normalised `fallback_name` casing to HA conventions.
* Added `unit="lx"` for the `SensorDeviceClass.ILLUMINANCE` number entity (`brightness_threshold`) to avoid HA warnings.
* Kept the hold-time entity named `fading_time` to match existing Tuya motion quirks.
* Removed the `translation_key` from the motion binary sensor, as the `MOTION` device class already provides the standard name.

Re: `brightness_threshold` — this datapoint controls the device’s internal ambient light threshold used for its built-in light/auto behaviour (i.e., whether the device should trigger its internal control logic based on current illuminance). Motion detection/reporting itself still functions; the threshold affects device-side automation logic rather than suppressing the ZHA motion entity.

Reviewed #4044 comments—kept naming consistent with existing Tuya motion patterns (notably fading_time), stuck to sentence-case fallback_name, and avoided extra translation keys where device_class already defines the standard entity name.”

TheJulianJES commented:
Did you push the changes to the branch for this PR?

ausfas committed:
Resolve tuya_motion.py merge conflict with upstream dev

pre-commit-ci[bot] committed:
Apply pre-commit auto fixes

ausfas committed:
Remove unrelated Tuya MCU change

ausfas commented:
@TheJulianJES , thanks.
I had accidentally pushed the update to the wrong local branch earlier. The requested changes have now been pushed to the PR branch (`add_ts0601_motion`).

ausfas committed:
Merge branch 'dev' into add_ts0601_motion

ausfas commented:
Hi @TheJulianJES,  just bumping this as all CI checks are passing successfully. Let me know if there is anything else needed on my end to get this merged. 
Thanks!
